src/traccuracy/matchers/_matched.py

```diff
@@ -1,45 +1,82 @@
-        The Matched objects will store both gt and pred data, as well as the mapping,
-        and any additional private attributes that may be needed/used e.g. detection matrices.
+    Each Matcher subclass will implement its own kwargs as needed.
+    In use, the Matcher object will be initialized with kwargs prior to running compute_mapping
+    on a particular dataset
+    """
+
+    def compute_mapping(self, gt_graph: "TrackingGraph", pred_graph: "TrackingGraph"):
+        """Run the matching on a given set of gt and pred TrackingGraph and returns a Matched object
+        with a new copy of each TrackingGraph
 
         Args:
             gt_graph (TrackingGraph): Tracking graph object for the gt
             pred_graph (TrackingGraph): Tracking graph object for the pred
+
+        Returns:
+            matched (Matched): Matched data object
+
+        Raises:
+            ValueError: gt and pred must be a TrackingGraph object
         """
-        self.gt_graph = gt_graph
-        self.pred_graph = pred_graph
+        if not isinstance(gt_graph, TrackingGraph) or not isinstance(
+            pred_graph, TrackingGraph
+        ):
+            raise ValueError(
+                "Input data must be a TrackingData object with a graph and segmentations"
+            )
 
-        self.mapping = self.compute_mapping()
+        matched = self._compute_mapping(gt_graph, pred_graph)
 
         # Report matching performance
-        total_gt = len(self.gt_graph.nodes())
-        matched_gt = len({m[0] for m in self.mapping})
-        total_pred = len(self.pred_graph.nodes())
-        matched_pred = len({m[1] for m in self.mapping})
+        total_gt = len(matched.gt_graph.nodes())
+        matched_gt = len({m[0] for m in matched.mapping})
+        total_pred = len(matched.pred_graph.nodes())
+        matched_pred = len({m[1] for m in matched.mapping})
         logger.info(f"Matched {matched_gt} out of {total_gt} ground truth nodes.")
         logger.info(f"Matched {matched_pred} out of {total_pred} predicted nodes.")
 
-    @abstractmethod
-    def compute_mapping(self):
-        """Computes a mapping of nodes in gt to nodes in pred
+        return matched
 
-        The mapping must be a list of tuples, e.g. [(gt_node, pred_node)]
+    @abstractmethod
+    def _compute_mapping(self, gt_graph: "TrackingGraph", pred_graph: "TrackingGraph"):
+        """Computes a mapping of nodes in gt to nodes in pred and returns a Matched object
 
         Raises:
             NotImplementedError
         """
         raise NotImplementedError
+
+
+class Matched:
+    """Matched data class which stores TrackingGraph objects for gt and pred
+    and the computed mapping
+
+    Each TrackingGraph will be a new copy on the original object
+
+    Args:
+        gt_graph (TrackingGraph): Tracking graph object for the gt
+        pred_graph (TrackingGraph): Tracking graph object for the pred
```

Comment:
note also, you're getting some errors in the docs build as well:

```
/home/docs/checkouts/readthedocs.org/user_builds/traccuracy/checkouts/103/docs/source/autoapi/traccuracy/matchers/_matched/index.rst:46: WARNING: more than one target found for cross-reference 'TrackingGraph': traccuracy._tracking_graph.TrackingGraph, traccuracy.TrackingGraph
/home/docs/checkouts/readthedocs.org/user_builds/traccuracy/checkouts/103/docs/source/autoapi/traccuracy/matchers/_matched/index.rst:46: WARNING: more than one target found for cross-reference 'TrackingGraph': traccuracy._tracking_graph.TrackingGraph, traccuracy.TrackingGraph
/home/docs/checkouts/readthedocs.org/user_builds/traccuracy/checkouts/103/docs/source/autoapi/traccuracy/matchers/_matched/index.rst:63: WARNING: more than one target found for cross-reference 'TrackingGraph': traccuracy._tracking_graph.TrackingGraph, traccuracy.TrackingGraph
/home/docs/checkouts/readthedocs.org/user_builds/traccuracy/checkouts/103/docs/source/autoapi/traccuracy/matchers/_matched/index.rst:63: WARNING: more than one target found for cross-reference 'TrackingGraph': traccuracy._tracking_graph.TrackingGraph, traccuracy.TrackingGraph
```

and I think it can be fixed by changing this to:

```suggestion
        gt_graph (traccuracy.TrackingGraph): Tracking graph object for the gt
        pred_graph (traccuracy.TrackingGraph .TrackingGraph): Tracking graph object for the pred
```

Reply:
Switching to "traccuracy.TrackingGraph" fixed the problem with the docs build, but I noticed in several other files we follow the same pattern of just using "TrackingGraph" in the docstring without any issues. Any idea why?

Reply:
> in several other files we follow the same pattern of just using "TrackingGraph" in the docstring without any issues

in the docstring?  or in the type hint?  this is somewhat tricky stuff.  For docstrings it depends on how sphinx is doing the type resolution from string to symbol.  just for the sake of debugging... and to clarify whether we're talking about docstrings or type hints, can you point to the module you noticed "TrackingGraph"

Reply:
Realistically as long as it's working I can live with the ambiguity 🤷 But here's the places I noticed it:
https://github.com/Janelia-Trackathon-2023/traccuracy/blob/7b4156280613216c70c7729026c5617912d609e3/src/traccuracy/matchers/_iou.py#L116-L117
https://github.com/Janelia-Trackathon-2023/traccuracy/blob/7b4156280613216c70c7729026c5617912d609e3/src/traccuracy/matchers/_ctc.py#L31-L32

Reply:
oh hell, those are indeed super annoying :joy:  I have no explanation!  stuck in the mysteries of sphinx at the moment.  I too am happy to live with the ambiguity if you are :)